File: gin/util/url.go
package util

import "net/url"

type Url struct{}

/* 编码 */
func (Url) Encode(data string) string {
	return url.QueryEscape(data)
}

/* 解码 */
func (Url) Decode(data string) string {
	res, err := url.QueryUnescape(data)
	if err != nil {
		return ""
	}
	return res
}

File: php/util/Url.php
<?php
namespace Util;

class Url {

  /* 编码 */
  static function Encode(string $data): string {
    return urlencode($data);
  }

  /* 解码 */
  static function Decode(string $data): string {
    return urldecode($data);
  }

}